src/js/loader-code.coffee

```diff
@@ -0,0 +1,627 @@
+#@codekit-prepend "picoModal"
+###
+Copyright (c) 2012 ElevenBlack
+Written by Ciocanel Razvan (chocksy.com)
+###
+
+###
+A self-contained loader library
+###
+window.widgetLoader = ((window,document) ->
+  "use strict"
+
+  defaults=
+    widget_domain:  '//location.for.iframe.widget' # the contact form to load
+    domain:         '//domain.for.iframe.widget'
+    modal_width:    false
+    modal_height:   false
+    iframe_widget:  false
+    iframe_width:   "100%"
+    iframe_height:  "100%"
+    side_btn:       true 
+
+  cssNumber=
+    "columnCount": true,
+    "fillOpacity": true,
+    "flexGrow": true,
+    "flexShrink": true,
+    "fontWeight": true,
+    "lineHeight": true,
+    "opacity": true,
+    "order": true,
+    "orphans": true,
+    "widows": true,
+    "zIndex": true,
+    "zoom": true
+
+  elements= 
+    side_btn_content:     '<div id="WDG_sideBtn_ctn"><a href="#" id="WDG_sideBtn">Errors Widget</a></div>'
+    side_btn:             "#WDG_sideBtn"
+
+  # ---- assignModal Method
+  # -- assign modal method to element class
+  assignModal= ->
+    $s('.el-modal').on 'click',(e)=>
+      e.preventDefault()
+      element = e.currentTarget 
+      widget_token = element.getAttribute('data-widget')
+      moduleInfo = JSON.stringify({url:widget_token})
+      loadModule({data:moduleInfo})
+
+  # ---- loadModule Method
+  # -- we use this method as a way to verify if we need to open a new window or a modal
+  # -- depending by the browser type. This is also called by the iframe in case of displaying
+  # -- a new step.
+  loadModule= (e)->
+    info_received = JSON.parse(e.data)
+    window.ELopts.widget_url = info_received.url
+    ###
+    window.ELopts.domain = window.ELopts.widget_domain
+    window.ELopts.domain = info_received.domain if info_received.domain!=undefined
+
+    if isMobile()
+      window.open(window.ELopts.domain+"?id="+window.ELopts.widget_url,'_blank')
+    else
+    ###  
+    openModal()
+
+  # ---- openModal Method
+  # -- we use this method to initialize the modal and add the iframe to it
+  openModal= ()->
+    current_height = make().getWindow('height')
+    current_width = make().getWindow('width')
+    widget_width = if window.ELopts.modal_width then window.ELopts.modal_width else current_width/1.2
+    widget_height = if window.ELopts.modal_height then window.ELopts.modal_height else current_height/1.6
+
+    outerWidth = if typeof widget_width=="number" then current_width-widget_width else (current_width*parseInt(widget_width)/100)
+    outerHeight= if typeof widget_height=="number" then current_height-widget_height else (current_height*parseInt(widget_height)/100)
+
+    picoModal(
+      content: '
+        <link rel="stylesheet" href="../src/css/epicloggerModal.css">
```

Comment:
@yokosogithub i'm not sure this will work. This js file will load into another website so i would not want to require them to add a css file too. Just a js file would be ideal. Try to see if we can move the css code in the js file also (inline of course). :D